src/easylink/pipeline_schema.py

```diff
@@ -54,6 +54,26 @@ def __init__(self, name: str, nodes: Iterable[Step], edges: Iterable[EdgeParams]
     def __repr__(self) -> str:
         return f"PipelineSchema.{self.name}"
 
+    def get_implementation_graph(self) -> ImplementationGraph:
```

Comment:
You'll find a common pattern throughout this PR. Whereas we used to call `get_implementation_graph()` on each step's ConfigurationState, we now instead explicitly call `add_{nodes,edges}_to_implementation_graph()`. This method is now the only time a `get_implementation_graph()` is defined (and it in turn simply adds nodes and adds edges).